tools/analysis_tools/bio3d_pca.xml

```diff
@@ -0,0 +1,157 @@
+<tool id="bio3d_PCA_analysis" name="PCA" version="0.1.2">
+    <description>Principle component analysis using Bio3D</description>
+    <requirements>
+          <requirement type="package" version="2.3_3">r-bio3d</requirement>
+    </requirements>
+    <stdio>
+        <exit_code range="1:" />
+    </stdio>
+    <command interpreter="Rscript">
+<![CDATA[ 
+    bio3d_pca.R 
+        $dcdin 
+        $pdbin
+        $method
+        #if $pca.sele == "calpha":
+        "string"
+        $pca.calpha
+        #end if
+        #if $pca.sele == "cbeta":
+        "string"
+        $pca.cbeta
+        #end if
+        #if $pca.sele == "backbone":
+        "string"
+        $pca.backbone
+        #end if
+        #if $pca.sele == "sidechain":
+        "string"
+        $pca.sidechain
+        #end if
+        #if $pca.sele == "protein":
+        "string"
+        $pca.protein
+        #end if
+        #if $pca.sele == "ligand":
+        "string"
+        $pca.ligand
+        #end if
+        #if $pca.sele == "nucleic":
+        "string"
+        $pca.nucleic
+        #end if
+        #if $pca.sele == "elety":
+        "elety"
+        $pca.elety
+        #end if
+        #if $pca.sele == "resid":
+        "resid"
+        $pca.resid
+        #end if
+        #if $pca.sele == "segid":
+        "segid"
+        $pca.segid
+        #end if
+        $output 
+        $pca_plot 
+        $pca_cluster 
+        $pc1_rmsf
+    2>&1
+]]></command>
+    <inputs>
+        <param format="dcd" name="dcdin" type="data" label="dcd trajectory input"/>
+        <param format="pdb" name="pdbin" type="data" label="pdb input"/>
+        <param name="method" type="boolean" truevalue="TRUE" falsevalue="FALSE" checked="false" label="Use singular value decomposition (SVD) instead of default eigenvalue decomposition ?" help="Default: No" />
+        <conditional name="pca">
+           <param name="sele" type="select" label="Select domains">
+             <option value="calpha">Calpha</option>
+             <option value="cbeta">Cbeta</option>
+             <option value="backbone">Backbone</option>
+             <option value="sidechain">Sidechain</option>
+             <option value="protein">Protein</option>
+             <option value="ligand">Ligand</option>
+             <option value="nucleic">Nucleic Acids</option>
+             <option value="elety">Atom Names</option>
+             <option value="resid">Resid</option>
+             <option value="segid">Segid</option>
+           </param>
+           <when value="calpha">
+              <param name="calpha" type="text" value="calpha"/>
```

Comment:
I don't understand this parameter. Is the aim that the user changes this text field? What are valid inputs?

Reply:
I will add help sections to each parameters. calpha is the alpha carbons of the protein backbone.

Reply:
@tsenapathi the question is, is the user supposed to change the text? If so to what?